### ISSUE TITLE
'Default access token' doesn't seem to allow link previews at all

### ISSUE THREAD

**Issue by mig5** [nextcloud/integration_github#84]
Hi

I want to enable link previews in Nextcloud documents for Github issues that are in private repositories.

I don't wish for my Nextcloud users to connect their own Github accounts to their Nextcloud accounts via OAuth.

I created a personal access token (the 'beta' kind, with finely scoped Read-Only access to: code, discussions, issues, metadata, and pull requests 

I added that PAT to the global administrative settings in Nextcloud under settings/admin/connected-accounts as the 'Default access token'. 

I ticked the box 'Enable Github link previews'

I still get 'Github API error Not Found' widgets appear in my documents when I paste a link to a Github issue in a private repository that this Github user has access to. The widget wants me to go to my personal account settings in Nextcloud to configure Github access, but I thought the 'default' access token should take care of it.

What's the problem? Is it the beta 'fine-grained' PAT method? I don't want to use the Classic method because I don't want Nextcloud to have write access to private repos through this token.

Otherwise, what are the scopes required for the PAT to work, with the beta method?

Do I still have to set up an OAuth2.0 app even though my use case is limited to link previews?

I am running Nextcloud 25.0.8, Github integration app 1.0.15

**Comment by julien-nc** [comment]
PAT are not able to get user notifications so the dashboard widget can't work with it. There will be an admin setting to globally disable the dashboard widget which can be useful if providing a global default access token.

I tried creating a PAT and leave the default permissions, I've set it as the default access token in the NC admin settings and I'm able to see link previews.

Keep in mind the link preview data is cached on the server side (one hour) and on the browser side (one hour as well but you can force-reload the page to get rid of the browser cache).
So if you really wanna test the link previews after changing some settings, provide a new link so you're sure the link preview data is not cached anywhere.

**Comment by julien-nc** [comment]
Yeah sorry, i was talking about a fine grained token. I just created it and didn't change anything (leave the "Repository access" to "Public Repositories (read-only)". Link previews to issues/PRs to public repos work fine with it (whether it's used by a user or as a "default access token" in NC admin settings). I then tried to check "All repositories" for this token and gave it the read-only access for Issues. Link previews work fine as well for links to issues in private repos.

Try using new GitHub links each time you make tests.

Also, I think you're right, classic GitHub access token can't provide read-only access to private repos. I tried a token which has "Full control of private repositories" and it had write permissions, but a token with no scope selected could not even read the repo (get info for a specific issue for example).
